src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java

```diff
@@ -574,6 +577,11 @@ public U updateInAnyTenant(U user) throws NotExistException {
     @Transactional(rollbackOn = Exception.class)
     public void saveUsers(List<U> users, Set<Tenant> tenants, boolean emailConfirmationRequired) {
 
+        Map<String, String> emailValidationErrors = getEmailValidationErrors(users);
```

Comment:
Wouldn't it be easier to just rethrow NotUniqueException with proper key for UI (and exception handler)?

Reply:
we need to gather all email field errors that's why NotUniqueException will not be useful